dids/src/main/kotlin/web5/sdk/dids/DidIonManager.kt

```diff
@@ -0,0 +1,356 @@
+import io.ktor.client.engine.cio.CIO
+import io.ktor.client.plugins.contentnegotiation.ContentNegotiation
+import io.ktor.client.request.get
+import io.ktor.client.request.post
+import io.ktor.client.request.setBody
+import io.ktor.client.statement.HttpResponse
+import io.ktor.client.statement.bodyAsText
+import io.ktor.http.ContentType
+import io.ktor.http.contentType
+import io.ktor.http.isSuccess
+import io.ktor.serialization.jackson.jackson
+import kotlinx.coroutines.runBlocking
+import org.erdtman.jcs.JsonCanonicalizer
+import org.erwinkok.multiformat.multicodec.Multicodec
+import org.erwinkok.multiformat.multihash.Multihash
+import org.erwinkok.result.get
+import org.erwinkok.result.getOrThrow
+import web5.sdk.common.Convert
+import web5.sdk.crypto.KeyManager
+import web5.sdk.dids.ion.model.Commitment
+import web5.sdk.dids.ion.model.Delta
+import web5.sdk.dids.ion.model.Document
+import web5.sdk.dids.ion.model.InitialState
+import web5.sdk.dids.ion.model.OperationSuffixDataObject
+import web5.sdk.dids.ion.model.PublicKey
+import web5.sdk.dids.ion.model.PublicKeyPurpose
+import web5.sdk.dids.ion.model.ReplaceAction
+import web5.sdk.dids.ion.model.SidetreeCreateOperation
+import java.util.UUID
+
+private const val operationsPath = "/operations"
+private const val identifiersPath = "/identifiers"
+
+/**
+ * Configuration for the [DidIonManager].
+ *
+ * @property ionHost The ION host URL.
+ * @property engine The engine to use. When absent, a new one will be created from the [CIO] factory.
+ */
+public class DidIonConfiguration internal constructor(
+  public var ionHost: String = "https://ion.tbddev.org",
+  public var engine: HttpClientEngine? = null,
+)
+
+
+/**
+ * Returns a [DidIonManager] after applying the provided configuration [builderAction].
+ */
+public fun DidIonManager(builderAction: DidIonConfiguration.() -> Unit): DidIonManager {
+  val conf = DidIonConfiguration().apply(builderAction)
+  return DidIonManagerImpl(conf)
+}
+
+/** [DidIonManager] is sealed, so we provide an impl so the constructor can be called. */
+private class DidIonManagerImpl(configuration: DidIonConfiguration) : DidIonManager(configuration)
+
+/**
+ * Provides a specific implementation for creating and resolving "did:ion" method Decentralized Identifiers (DIDs).
+ *
+ * A "did:ion" DID is an implementation of the Sidetree protocol that uses Bitcoin as it's anchoring system.
+ * Further specifics and technical details are outlined in [the DID Sidetree Spec](https://identity.foundation/sidetree/spec/).
+ *
+ * @property uri The URI of the "did:ion" which conforms to the DID standard.
+ * @property keyManager A [KeyManager] instance utilized to manage the cryptographic keys associated with the DID.
+ * @property creationMetadata Metadata related to the creation of a DID. Useful for debugging purposes.
+ *
+ * ### Usage Example:
+ * ```kotlin
+ * val keyManager = InMemoryKeyManager()
+ * val did = DidKey("did:key:example", keyManager)
+ * ```
+ */
```

Comment:
update example to show `DidIon` instead of `DidKey`

Reply:
Done.